modules/custom/az_digital_asset_library/az_digital_asset_library.module

```diff
@@ -6,51 +6,52 @@
-use Drupal\Core\Url;
+use Drupal\Component\Serialization\Json;
 
 /**
- * Custom submit handler to redirect to /admin/content/media.
- */
-function az_digital_asset_library_form_submit(array $form, FormStateInterface $form_state) {
-  $url = Url::fromRoute('entity.media.collection');
-  $form_state->setRedirectUrl($url);
-}
-
-/**
- * Implements hook_FORM_ID_alter().
+ * Implements hook_preprocess_HOOK().
  *
- * Add custom submit handler to form.
- * Allow using remote media.
- * See
- * @link https://git.drupalcode.org/project/intelligencebank/-/blob/8.x-2.x/modules/ib_dam_media/src/Plugin/EntityBrowser/Widget/IbDamEbWidget.php#L249 @endlink.
+ * Hide default media type provided by IntelligenceBank from media add page.
+ *
+ * See @link https://github.com/az-digital/az_quickstart/issues/1741
  */
-function az_digital_asset_library_form_entity_browser_az_digital_asset_library_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  $form['#submit'][] = 'az_digital_asset_library_form_submit';
+function az_digital_asset_library_preprocess_entity_add_list(&$variables) {
+  if (isset($variables['bundles']['ib_dam_embed'])) {
+    unset($variables['bundles']['ib_dam_embed']);
+  }
 }
 
 /**
- * Implements hook_preprocess_HOOK().
+ * Implements hook_form_FORM_ID_alter().
  *
- * Hide default media type provided by IntelligenceBank from media select
- * dialog.
+ * Change label text for open browser button.
+ * Change dialog title option.
  *
- * See @link https://github.com/az-digital/az_quickstart/issues/1741
+ * See @link drupal.org/project/intelligencebank/issues/3353858
  */
-function az_digital_asset_library_preprocess_links__media_library_menu(&$variables) {
-  if (isset($variables['links']['media-library-menu-ib_dam_embed']['link'])) {
-    $variables['links']['media-library-menu-ib_dam_embed']['link']['#access'] = FALSE;
+function az_digital_asset_library_form_media_library_add_form_upload_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $data_dialog_options = [];
+  $form['ib_dam_container']['asset_browser']['#title'] = t('Add Arizona Digital Asset Library Media');
+  if (isset($form['ib_dam_container']['asset_browser']['#attributes']['data-dialog-options'])) {
+    $data_dialog_options = Json::decode($form['ib_dam_container']['asset_browser']['#attributes']['data-dialog-options']);
   }
+  $data_dialog_options['title'] = 'Arizona Digital Asset Library Browser';
+  $form['ib_dam_container']['asset_browser']['#attributes']['data-dialog-options'] = Json::encode($data_dialog_options);
 }
 
 /**
- * Implements hook_preprocess_HOOK().
+ * Implements hook_module_implements_alter().
  *
- * Hide default media type provided by IntelligenceBank from media add page.
+ * Move az_digital_asset_library to the end of the list of modules that
+ * implement hook_form_alter.
  *
- * See @link https://github.com/az-digital/az_quickstart/issues/1741
+ * This could possibly be done in az_digital_asset_library.install, but it's
```

Comment:
Should we remove these comments before merging?

Reply:
I think we can leave them as future improvement notes.

Reply:
I think they're worth preserving.